src/main/java/emissary/util/ShortNameComparator.java

```diff
@@ -19,34 +19,115 @@
     public int compare(IBaseDataObject obj1, IBaseDataObject obj2) {
         String s1 = obj1.shortName();
         String s2 = obj2.shortName();
+        int index1 = s1.indexOf(Family.SEP, 0);
+        int index2 = s2.indexOf(Family.SEP, 0);
 
-        String[] p1 = s1.split(Family.SEP);
-        String[] p2 = s2.split(Family.SEP);
-
-        for (int i = 1; i < p1.length; i++) {
-
-            if (i >= p2.length) {
+        while (index1 >= 0) {
+            if (index1 >= 0 && index2 < 0) {
                 return 1;
             }
 
-            if (p1[i].equals(p2[i])) {
+            index1 += Family.SEP.length();
+            index2 += Family.SEP.length();
+
+            final int nextIndex1 = s1.indexOf(Family.SEP, index1);
+            final int nextIndex2 = s2.indexOf(Family.SEP, index2);
+            final int length1 = ((nextIndex1 < 0) ? s1.length() : nextIndex1) - index1;
+            final int length2 = ((nextIndex2 < 0) ? s2.length() : nextIndex2) - index2;
+
+            if (length1 == length2 && s1.regionMatches(index1, s2, index2, length1)) {
+                index1 = nextIndex1;
+                index2 = nextIndex2;
+
                 continue;
             }
 
             try {
-                int i1 = Integer.parseInt(p1[i]);
-                int i2 = Integer.parseInt(p2[i]);
-                return i1 - i2;
+                int int1 = parseInt(s1, 10, index1, length1);
+                int int2 = parseInt(s2, 10, index2, length2);
+
+                return int1 - int2;
             } catch (NumberFormatException e) {
-                return p1[i].compareTo(p2[i]);
-            }
+                final String substring1 = s1.substring(index1, index1 + length1);
```

Comment:
Comment something akin to: "If numeric comparison fails, perform a string comparison"